packages/block-library/src/tag-cloud/edit.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { map, filter } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import {
+	PanelBody,
+	Placeholder,
+	ToggleControl,
+	SelectControl,
+	ServerSideRender,
+} from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { InspectorControls } from '@wordpress/editor';
+
+class TagCloudEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.setTaxonomy = this.setTaxonomy.bind( this );
+		this.toggleShowTagCounts = this.toggleShowTagCounts.bind( this );
+	}
+
+	getTaxonomyOptions() {
+		const taxonomies = filter( this.props.taxonomies, 'show_cloud' );
+		const selectOption = {
+			label: __( '- Select -' ),
+			value: '',
+		};
+		const taxonomyOptions = map( taxonomies, ( taxonomy ) => {
+			return {
+				value: taxonomy.slug,
+				label: taxonomy.name,
+			};
+		} );
+
+		return [ selectOption, ...taxonomyOptions ];
+	}
+
+	setTaxonomy( taxonomy ) {
+		const { setAttributes } = this.props;
+
+		setAttributes( { taxonomy } );
+	}
+
+	toggleShowTagCounts() {
+		const { attributes, setAttributes } = this.props;
+		const { showTagCounts } = attributes;
+
+		setAttributes( { showTagCounts: ! showTagCounts } );
+	}
+
+	render() {
+		const { attributes } = this.props;
+		const { taxonomy, showTagCounts } = attributes;
+		const taxonomyOptions = this.getTaxonomyOptions();
+
+		const inspectorControls = (
+			<InspectorControls>
+				<PanelBody title={ __( 'Tag Cloud Settings' ) }>
+					<SelectControl
+						label={ __( 'Taxonomy' ) }
+						options={ taxonomyOptions }
+						value={ taxonomy }
+						onChange={ this.setTaxonomy }
+					/>
+					<ToggleControl
+						label={ __( 'Show tag counts' ) }
+						checked={ showTagCounts }
+						onChange={ this.toggleShowTagCounts }
+					/>
+				</PanelBody>
+			</InspectorControls>
+		);
+
+		if ( ! taxonomy ) {
+			return (
+				<Fragment>
+					{ inspectorControls }
+					<Placeholder
+						key="placeholder"
+						icon="tag"
+						label={ __( 'Tag Cloud' ) }
+						instructions={ __( 'Select a Taxonomy' ) }
+					>
+						<SelectControl
+							options={ taxonomyOptions }
+							onChange={ this.setTaxonomy }
+						/>
```

Comment:
This select element misses a properly associated `<label>` element. Should use a prop
`label={ __( 'Select a Taxonomy' ) }` 
And the `instructions` prop should be removed from the placeholder.